test/metabase/db/force_migration_test.clj

```diff
@@ -38,13 +38,12 @@
     (mt/with-temp-empty-app-db [conn driver/*driver*]
       (with-redefs [liquibase/changelog-file "force-migration.yaml"]
         (let [data-source  (mdb/data-source)
-              database     (->> (if (instance? java.sql.Connection conn)
```

Comment:
Do we understand why this overloading was necessary before, to be sure we can remove it?

Just having a Chesterton's fence feeling, as unlikely as it is.